cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/RestartContext.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.resource;
+
+import io.strimzi.operator.cluster.model.RestartReasons;
+import io.strimzi.operator.common.BackOff;
+import io.vertx.core.Promise;
+
+import java.util.function.Supplier;
+
+/**
+ * Contains state, flags as to what kind of restart (if any) to do, and reasons why
+ */
+class RestartContext {
```

Comment:
If I get it right, it is used only for Kafka restarts? So maybe `KafkaRestartContext` would be better?

Reply:
Good call :+1: